strawberryfields/backends/states.py

```diff
@@ -1250,6 +1252,19 @@ def number_expectation(self, modes):
             "The number_expectation method only supports one or two modes for Gaussian states."
         )
 
+    def parity_expectation(self, modes):
+        """Calculates the expectation value of a product of parity operators acting on given modes"""
+        if len(modes) != len(set(modes)):
+            raise ValueError("There can be no duplicates in the modes specified.")
+
+        mu = self._mu
```

Comment:
I think there a methods to access this hidden attributes; you should try to use them,